File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:

        if x < 0:
            return False
        
        original = x
        rev = 0

        while(x > 0):
            lastDigit = x % 10
            rev = rev * 10 + lastDigit
            x = x//10
        
        return rev == original
        

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        

        if (x < 0) return false;
        int og = x;

        int rev = 0;
        while(x > 0) {
            int lastDigit = x%10;
            rev = rev * 10 + lastDigit;
            x = x/10;
        }

        return og == rev;
    }
}